src/Microsoft.ML.Data/Training/TrainerEstimatorBase.cs

```diff
@@ -158,7 +158,7 @@ protected virtual void CheckLabelCompatible(SchemaShape.Column labelCol)
         protected virtual RoleMappedData MakeRoles(IDataView data) =>
             new RoleMappedData(data, label: LabelColumn?.Name, feature: FeatureColumn.Name, weight: WeightColumn?.Name);
 
-        IPredictor ITrainer.Train(TrainContext context) => ((ITrainer<TModel>)this).Train(context);
+        IPredictor ITrainer.Train(TrainContext context) => (IPredictor)((ITrainer<TModel>)this).Train(context);
```

Comment:
Yep, leftover from trying to extend the TrainerEstimatorBase for the CalibratorEstimators. Decided not to go for it, since that extends ITraner, but might make sense to do so. 

---
In reply to: [238803551](https://github.com/dotnet/machinelearning/pull/1816#discussion_r238803551) [](ancestors = 238803551)

Reply:
So I feel like increasing training estimators" to such a large extent basically makes them indistinguishable from general estimators, and I wonder if we should not do that. Is there a cost to having a separate type for training calibrators, while maintaining the integrity of the trainer estimators?